### PR TITLE
Cancel payment request

### PR DESCRIPTION
**Pull request by ofux**
- ✨ add cancel() method to payment aggregate
- ✨ add cancel_payment_request mutation
- ✅ add e2e tests

### CODE REVIEW

File: backend/event-listeners/src/domain/projectors/payment_request.rs [hunk 1]
```diff
@@ -22,24 +23,31 @@ impl Projector {
...
+						*id,
+						*budget_id,
+						*requestor_id,
+						*recipient_id,
+						*amount_in_usd as i64,
+						reason.clone(),
+					))
+					.map_err(DatabaseError::into),
+				PaymentEvent::Cancelled { id } =>
+					self.repository.delete(id).map_err(DatabaseError::into),
```
Comment (AnthonyBuisset) [inline backend/event-listeners/src/domain/projectors/payment_request.rs:47]: we should remove the foreign key between `payments` and `payment_requests` as we can submit "partial" payment receipts

File: backend/api/src/application/payment/cancel.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		let payment = self.payment_repository.find_by_id(&payment_id)?;
+		let events: Vec<_> = payment
+			.cancel()
+			.map_err(|e| DomainError::InvalidInputs(e.into()))?
+			.into_iter()
+			.map(Event::from)
+			.map(UniqueMessage::new)
+			.collect();
+
+		self.event_publisher
```
Comment (AnthonyBuisset) [inline backend/api/src/application/payment/cancel.rs:38]: if you import `Publishable`, you can chain everything with `.publish()` function

**Review by AnthonyBuisset — APPROVED** [review]

File: backend/event-listeners/src/domain/projectors/payment_request.rs (hunk 1 above)
Comment (ofux) [reply backend/event-listeners/src/domain/projectors/payment_request.rs:47]: For now, it is forbidden to cancel a payment as soon as a receipt has been submitted for it (even partial).
I think supporting such things require a proper specification as there are many possibilities with different implications for the user.

Comment (AnthonyBuisset) [reply backend/event-listeners/src/domain/projectors/payment_request.rs:47]: 👍

File: backend/api/src/application/payment/cancel.rs (hunk 2 above)
Comment (ofux) [reply backend/api/src/application/payment/cancel.rs]: done

**Pull request merged**